src/input_context/input_condition/pulse.rs

```diff
@@ -152,6 +153,45 @@ mod tests {
         );
     }
 
+    #[test]
+    fn pulse_fires_again_after_release() {
```

Comment:
Since it's inside `pulse` module, I prefer to shorten test names. Yes, we have a `pulse` test above, but I couldn't come up with a nicer name for it :sweat_smile: 
```suggestion
    fn fires_again_after_release() {
```

Reply:
ah, naming... 🫠
you are the boss 😂